File: module6/bai_thi_fe/product/src/app/model/product.ts
import {Category} from './category';

export interface Product {
  id?: number;
  code?: string;
  name?: string;
  price?: number;
  status?: string;
  importDate?: string;
  isDeleted?: boolean;
  category?: Category;

  made?: string;
  gas?: boolean;
  sweat?: boolean;
  author?: string;
  typeBook?: string;
  publicationDate?: string;

}

File: module6/bai_thi/product/src/main/java/com/example/model/product/Product.java
package com.example.model.product;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Getter
@Setter
@RequiredArgsConstructor
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String code;
    private String name;
    private String price;
    private String status;
    private String importDate;
    @Column(columnDefinition = "bit(1) default 0")
    private Boolean isDeleted;

    @ManyToOne()
    @JoinColumn(name = "category_id", referencedColumnName = "id")
    private Category category;

    private String made;
    private Boolean gas;
    private Boolean sweat;
    private String author;
    private String typeBook;
    private String publicationDate;

}
